pkg/archive/archive.go

```diff
@@ -116,7 +116,29 @@ func DetectCompression(source []byte) Compression {
 	return Uncompressed
 }
 
-func xzDecompress(archive io.Reader) (io.ReadCloser, error) {
+type wrapper struct {
+	r      io.ReadCloser
+	chdone <-chan struct{}
+}
+
+func (w *wrapper) Read(p []byte) (n int, err error) {
+	return w.r.Read(p)
+}
+
+func (w *wrapper) Close() error {
```

Comment:
This close function is a little weird, and seems like it would cause problems on multiple calls to close.
From what I can tell in the other code, you are intending to make sure stdin is done before the reader is closed, but return immediately?